packages/node-core/src/indexer/store.service.ts

```diff
@@ -282,23 +290,57 @@ export class StoreService {
   }
 
   async getHistoricalStateEnabled(): Promise<boolean> {
-    let enabled = true;
+    const {multiChain} = this.config;
+    let {disableHistorical} = this.config;
+
+    if (multiChain && !disableHistorical) {
+      logger.info('Historical state is not compatible with multi chain indexing, disabling historical..');
+      disableHistorical = true;
+    }
```

Comment:
The disableHistorical flag only applies with clean projects. e.g if you had already indexed historical and then disable through the flag it will still be enabled because the database is setup for it.

We should retain this behaviour, if the project already exists then historical is enabled based on the db schema.

With this current behaviour if multiChain is enabled and there is historical data its going to cause issues with the DB.